### PR TITLE
add predefined macros support in kafka engine settings

### PR DESCRIPTION
**Pull request by ybogo**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
add support of predefined macro (`{database}` and `{table}`) in some kafka engine settings: topic, consumer, client_id, etc.

If you want to use database name or table name as a part of your kafka topic name you use such query:
```
CREATE TABLE statistics_queue
(
    `request_id`        String
)
ENGINE = Kafka
SETTINGS kafka_broker_list = 'kafka',
kafka_topic_list = '{table}_statistics',
kafka_group_name = '{database}_group',
kafka_format = 'JSONEachRow',
kafka_row_delimiter = '\n',
kafka_num_consumers = 2,
kafka_skip_broken_messages = 0,
kafka_handle_error_mode = 'stream';
```

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by ybogo** [comment]
It is my first PR to ClickHouse. I added a test for my improvement but I can't find info how to run this test. So for now it may fail

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit d54c0d2434b67cf2e2b19cafae8609893f3fca1c with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52386/d54c0d2434b67cf2e2b19cafae8609893f3fca1c/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by antaljanosbenjamin — COMMENTED** [review]
I think the PR looks good, only minor comments.

File: tests/integration/test_storage_kafka/test.py [hunk 1]
```diff
@@ -444,6 +444,48 @@ def test_kafka_settings_new_syntax(kafka_cluster):
...
+    )
+
+    messages = []
+    for i in range(25):
+        messages.append(json.dumps({"key": i, "value": i}))
+    kafka_produce(kafka_cluster, "test_kafka", messages)
+
+    # Insert couple of malformed messages.
+    kafka_produce(kafka_cluster, "test_kafka", ["}{very_broken_message,"])
+    kafka_produce(kafka_cluster, "test_kafka", ["}another{very_broken_message,"])
```
[the remark this answers is not in the document]
Comment (ybogo) [reply tests/integration/test_storage_kafka/test.py]: Yes, they don't increase the value of test. It is just a copy-paste issue. Already removed

File: tests/integration/test_storage_kafka/test.py [hunk 2]
```diff
@@ -444,6 +444,48 @@ def test_kafka_settings_new_syntax(kafka_cluster):
...
+
+    result = ""
+    while True:
+        result += instance.query("SELECT * FROM test.kafka", ignore_error=True)
+        if kafka_check_result(result):
+            break
+
+    kafka_check_result(result, True)
+
+    members = describe_consumer_group(kafka_cluster, "new")
```
[the remark this answers is not in the document]
Comment (ybogo) [reply tests/integration/test_storage_kafka/test.py]: II managed to run the tests and reproduced this error. Already fixed

**Comment by ybogo** [comment]
@antaljanosbenjamin thanks for your review. I made all the changes you requested and managed to run integration tests. But I still have two questions.

1. I fixed style check with `clang-format` and CI successfully passed but it changed a lot of other lines inside `src/Storages/Kafka/StorageKafka.cpp` file (mostly if changed indents in some blocks of code). Is it okay or should I fix only code style only in my changes?
2. I realized that `Common/Macros.h` header was added in `src/Storages/Kafka/StorageKafka.h` public interface. Is it okay or should I add only forward declaration inside `StorageKafka` private section?

**Comment by antaljanosbenjamin** [comment]
> I fixed style check with clang-format and CI successfully passed but it changed a lot of other lines inside src/Storages/Kafka/StorageKafka.cpp file (mostly if changed indents in some blocks of code). Is it okay or should I fix only code style only in my changes?

Please only change the formatting for the lines that you changed. Feel free to check out [our docs](https://clickhouse.com/docs/en/development/tests#code-style) about code style. I can highly recommend to use `.clang-format`.

In general only change the lines that are necessary, so no reordering header etc, because it can really mess up the commit history.

> I realized that Common/Macros.h header was added in src/Storages/Kafka/StorageKafka.h public interface. Is it okay or should I add only forward declaration inside StorageKafka private section?

I think it is tricky to add forward declaration of a struct that supposed to be used as a member. As we don't generate forward headers, it is fine to include `Common/Macros.h`.

**Review by antaljanosbenjamin — APPROVED** [review]
The PR looks really good, thanks for the contribution!

In the tests the only thing that is not tested is the `schema_name`, but I think that is okay, as
1. the logic is not complex
2. The others where the same function is used is tested
3. There are tests for handling the schema

So it looks good to me. The test failures seems unrelated, but let me check them out quickly.

**Comment by antaljanosbenjamin** [comment]
Test failures:
- `00900_long_parquet_load`: issue #52863 created
- `02352_rwlock`: PR #52852

**Pull request merged**